pages/_app.js

```diff
@@ -1,10 +1,32 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
+import { transactions } from "@/db/data.js";
+import { uid } from "uid";
+import { useRouter } from "next/router";
+
+const initialTransactions = transactions;
 
 export default function App({ Component, pageProps }) {
+  const [transactions, setTransactions] = useState(initialTransactions);
+  const router = useRouter();
+
+  function handleAddTransaction(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newTransaction = Object.fromEntries(formData);
+    setTransactions([{ id: uid(), ...newTransaction }, ...transactions]);
+    event.target.reset();
+    router.push("/");
+  }
```

Comment:
Something that came to our mind was to split this function into two separate functions to make the code a bit more clean:
You could leave Line 17 at _app.js as a stand-alone function and move Line 14-16 + Line 18-19 as a separate function into TransactionEntryForm.js! Reason: This solution would guarantee that as much as possible from the functionality regarding the Form is placed at the Form component.